maven-model-builder/src/main/java/org/apache/maven/model/profile/DefaultProfileSelector.java

```diff
@@ -76,16 +85,24 @@ public List<org.apache.maven.api.model.Profile> getActiveProfilesV4(
     @Override
     public List<Profile> getActiveProfiles(
             Collection<Profile> profiles, ProfileActivationContext context, ModelProblemCollector problems) {
+
+        if (profiles.stream().map(Profile::getId).distinct().count() < profiles.size()) {
```

Comment:
What is this test about ? That seems unrelated to this PR.  Should this be rather validated (with a warning printed) earlier in the process ? During settings validation and project validation ?